source/base/parameter_handler.cc

```diff
@@ -1329,285 +1339,270 @@ ParameterHandler::recursively_print_parameters(
+                    out << "\\label{"
+                        << Utilities::replace_in_string(label, "_20", " ")
+                        << "}\n";
+                  out << "\\label{" << label << "}\n";
+                }
+                out << "\n\n";
 
-          break;
+                out << "\\index[prmindex]{" << escape(demangle(p.first))
+                    << "}\n";
+                out << "\\index[prmindexfull]{";
+                for (const auto &path : target_subsection_path)
+                  out << escape(path) << "!";
+                out << escape(demangle(p.first)) << "}\n";
+
+                // finally print alias and indicate if it is deprecated
+                out
+                  << "This parameter is an alias for the parameter ``\\texttt{"
+                  << escape(alias) << "}''."
+                  << (p.second.get<std::string>("deprecation_status") ==
+                          "true" ?
+                        " Its use is deprecated." :
+                        "")
+                  << "\n\n"
+                  << '\n';
+              }
+          out << "\\end{itemize}" << '\n';
         }
+    }
+  else if (style & Description)
+    {
+      // first find out the longest entry name to be able to align the
+      // equal signs
+      std::size_t longest_name = 0;
+      for (const auto &p : current_section)
+        if (is_parameter_node(p.second) == true)
+          longest_name = std::max(longest_name, demangle(p.first).length());
+
+      // print entries one by one
+      for (const auto &p : current_section)
+        if (is_parameter_node(p.second) == true)
+          {
+            // print name and value
+            out << std::setw(overall_indent_level * 2) << ""
+                << "set " << demangle(p.first)
+                << std::setw(longest_name - demangle(p.first).length() + 1)
+                << " "
+                << " = ";
+
+            // print possible values:
+            const unsigned int pattern_index =
+              p.second.get<unsigned int>("pattern");
+            const std::string full_desc_str =
+              patterns[pattern_index]->description(Patterns::PatternBase::Text);
+            const std::vector<std::string> description_str =
+              Utilities::break_text_into_lines(
+                full_desc_str, 78 - overall_indent_level * 2 - 2, '|');
+            if (description_str.size() > 1)
+              {
+                out << '\n';
+                for (const auto &description : description_str)
+                  out << std::setw(overall_indent_level * 2 + 6) << ""
+                      << description << '\n';
+              }
+            else if (description_str.empty() == false)
+              out << "  " << description_str[0] << '\n';
+            else
+              out << '\n';
 
-      default:
-        Assert(false, ExcNotImplemented());
+            // if there is a documenting string, print it as well
```

Comment:
Can you surround this in a `if(!(style & Short))`? This way we'll have also the `ShortDescription` implemented.

Also add this to the possible combinations, and fix the table at the beginning.